Sources/AsyncConnection.swift

```diff
@@ -3,7 +3,7 @@ public protocol AsyncConnection: AsyncStream {
 }
 
 extension AsyncConnection {
-    public func open(result: (Void throws -> AsyncConnection) -> Void) throws {
+    public func open(_ result: (Void throws -> AsyncConnection) -> Void) throws {
```

Comment:
As @paulofaria mentioned, argument label must describe role, not the type. `callback` is reasonable because it will be called after the completion of needed action, no matter if that was successful. For example, `NSURLSession` takes `completionHandler`, which gets called in every possible case, including for errors.
So, what I'm saying is that `completion` describes the role of the argument, and from its type it's clear that it's a closure that wraps up another closure. And `typealias`, I am sure, will only make it much less readable.